migrate/016_storage_encryption.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:cipher) do
```

Comment:
Probably should have a different name for this, it's a bit broad. I'm a little puzzled as to why it's a separate table from the vm_storage_volume.

Reply:
Initially I made this a separate table because I didn't like `vm_storage_volume` who having 4 columns who are "nullable" but their nullness should be equal & their names are prefixed with `key_encryption_`. Then I thought there's nothing specific here about storage encryption & this has pretty much all fields necessary to create OpenSSL::Cipher which can be used for other purposes too, so I named it `cipher`.

I can embed this in `vm_storage_volume`, no super-strong opinions here. What are your thoughts given above?

Reply:
ack re: nulls but I really doubt how much advisable re-use we'll get out of the cipher table. Also, don't you need a second set of fields somewhere for key rotation? I'm thinking about faults during rotation, where it's ambiguous whether the KEK has been rotated or not. Is there some kind of authenticated encryption (e.g. AEAD) that can be used to confirm whether a key is the valid one?

Reply:
I changed the table name, implemented key rotation, and updated the schema to fit the rotation algorithm.